db/migrate/20191125151240_migrate_profile_data.rb

```diff
@@ -0,0 +1,23 @@
+class MigrateProfileData < ActiveRecord::Migration[5.2]
+  def up
+    User.all.each do |user|
+      next if user.identity.present?
+      next if user.provider.blank?
```

Comment:
This will cause all users to be skipped, because it's using your updated `#provider` method which delegates to `Identity`.

Reply:
I think we can just kill this line?  If `user.identity.present?` is false then we always want to migrate the user, right?

Reply:
I was wrong - that will migrate test users.  Found a better way and pushed a new commit to fix.